src/pathfinding/grid_pathfinder.jl

```diff
@@ -0,0 +1,362 @@
+export CostMetric,
+    DirectDistance,
+    Chebyshev,
+    HeightMap,
+    AStar,
+    delta_cost,
+    find_path,
+    set_target!,
+    is_stationary,
+    heightmap,
+    walkmap
+
+"""
+    Path{D}
+Alias of `MutableLinkedList{Dims{D}}`. Used to represent the path to be
+taken by an agent in a `D` dimensional [`GridSpace{D}`](@ref).
+"""
+const Path{D} = MutableLinkedList{Dims{D}}
+
+"""
+    CostMetric{D}
+An abstract type representing a metric that measures the approximate cost of travelling
+between two points in a `D` dimensional [`GridSpace{D}`](@ref).
+"""
+abstract type CostMetric{D} end
+
+struct DirectDistance{D} <: CostMetric{D}
+    direction_costs::Vector{Int}
+end
+
+Base.show(io::IO, metric::DirectDistance) = print(io, "DirectDistance")
+
+"""
+    DirectDistance
+Distance is approximated as the shortest path between the two points, provided the
+`walkable` property of [`AStar`](@ref) allows.
+Optionally provide a `Vector{Int}` that represents the cost of going from a tile to the
+neighboring tile on the `i` dimensional diagonal (default is `10√i`).
+
+If `moore_neighbors=false` in [`AStar`](@ref), only Von Neumann neighbors will be tested.
+Cost defaults to the first value of the provided vector.
+"""
+DirectDistance{D}() where {D} = DirectDistance{D}([floor(Int, 10.0 * √x) for x in 1:D])
+
+"""
+    Chebyshev
+Distance between two tiles is approximated as the Chebyshev distance (maximum of absolute
+difference in coordinates) between them.
+"""
+struct Chebyshev{D} <: CostMetric{D} end
```

Comment:
... You can't do this. Distances.jl defines and exports `Chebyshev`. And we use this same type in `GridSpace`. I'm not even sure how defining this didn't break everything. Why don't you simply re-use the existing `Distances.Chebyshev` or even better, create a different name, like `MaxDistance`.

Reply:
We don't have `Distances` in our deps, so I don't see how this is an issue. Maybe `MaxDistance` is fine though, if are planning to rename moore and vonneumann to something less precise/technical.

Reply:
Distances.jl is a very popular package, if not for users definitely for developers. I'm not comfortable using identical names to it without them coming directly from it. Let's go with `MaxDistance` or just check if `Distances` is already in our dependencies as a secondary dependency (extremely likely) and import the name from there.

Reply:
It's not in our dependencies at all, so we should do `MaxDistance`.